ui/src/components/SampleView/DrawGridPlugin.js

```diff
@@ -593,19 +593,31 @@ export default class DrawGridPlugin {
     }
   }
 
-  getClickedCell(shapeGroup, clickPoint) {
-    let cell = null;
+  getClickedCellIndex(gd, shapeGroup, e) {
+    const cellSizeX = this.getCellWidth(gd);
+    const cellSizeY = this.getCellHeight(gd);
 
-    shapeGroup.forEachObject((obj) => {
-      if (
-        obj.get('type') === 'ellipse' &&
-        obj.containsPoint(clickPoint, null, true)
-      ) {
-        cell = obj;
-      }
-    });
+    const cellIdxX = Number.parseInt(
```

Comment:
Not quite sure why you need `parseInt`.

Reply:
not sure, looks indeed strange. Ill have a look